core/module.h

```diff
@@ -18,6 +19,14 @@ using bess::gate_idx_t;
 #define MAX_NUMA_NODE 16
 #define UNCONSTRAINED_SOCKET ((0x1ull << MAX_NUMA_NODE) - 1)
 
+class Node {
+ public:
+  Node(Module *module) : module_(module), children_(), parents_() {}
+  Module *module_;
+  std::unordered_set<std::string> children_;
+  std::unordered_set<Module *> parents_;
+};
```

Comment:
Do you need Node to be copyable?  If not, use DISALLOW_COPY_AND_ASSIGN(Node) in the private section or delete the constructors.